TensorMath.lua

```diff
@@ -51,6 +51,18 @@ static int torch_isnonemptytable(lua_State *L, int idx)
 }
 ]])
 
+
+interface:print([[
+static const void* torch_istensorarray(lua_State *L, int idx)
+{
+  if (!torch_isnonemptytable(L, idx)) return 0;
+
+  lua_checkstack(L, 1);
+  lua_rawgeti(L, 1, 1);
+  return (torch_istensortype(L, luaT_typename(L, -1)));
```

Comment:
Yes, it should be `lua_rawgeti(L, idx, 1)`. I did inspect the values on the stack to make sure the value of lua_rawgeti was deleted with a stackDump function, and in all cases this seemed to work, but this is probably better.